app/services/guest-cam/index.ts

```diff
@@ -639,24 +659,26 @@ export class GuestCamService extends StatefulService<IGuestCamServiceState> {
 
     this.log('New guest joined', event);
 
-    this.SET_GUEST({ name: event.data.name, streamId: event.data.streamId });
+    // If we have a vacant source, assign the guest now
+    const vacantSources = this.views.vacantSources;
+    const sourceId = vacantSources.length ? vacantSources[0].sourceId : undefined;
 
-    // Clean up any existing consumer before connecting the new one
-    if (this.consumer) {
-      this.consumer.destroy();
-    }
+    this.ADD_GUEST({ remoteProducer: event.data, sourceId, showOnStream: false });
```

Comment:
if i'm understanding this part right the first guest that joins will fill the first available slot? does that mean it's impossible to have sources tied to guests? could be a bit inconvenient for regular collaborators who have certain areas of the stream blocked for specific people

Reply:
That's correct.  By default we assign people to slots in the order they come in.  However, guests are hidden until you show them, and included in this PR is the ability to swap guests between sources.  So I'm imagining that if you have a specific slot set up for a specific person, you would just swap the source they were assigned to before showing them on your stream.

Long term maybe we could figure out a way to "remember" this for regular guests, but I'm not entirely sure what this would like.

Reply:
gotcha! sounds good, didn't realize there was a swapping function so i think that should be good enough as long as it's obvious